packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1033,6 +1079,15 @@ export class Combobox
       largestSelectedIndicatorChipWidth,
     });
 
+    if (this.isSelectAllOptionChecked) {
+      this.selectedItems.forEach((item) => {
+        const chipEl = this.referenceEl.querySelector<Chip["el"]>(`#${chipUidPrefix}${item.guid}`);
```

Comment:
Can we get reference to this chip el without querying the DOM?

Reply:
> Can we get reference to this chip el without querying the DOM?

Don't think so since the chips are dynamically generated based on the combobox-items.  I think this applies to your other questions about matching elements instead of ids for the arrays of items.  We can certainly set a ref on the select all checkbox element, but not sure if it makes sense to do this for the elements.  The refs would have to be garbage collected if the light dom children are modified.

Reply:
Yeah I just meant the elements that the component is creating, not user slotted ones.

Reply:
I'll change the other 2, but leave this one alone then. Since it generates the chips for slotted items.

Reply:
> Yeah I just meant the elements that the component is creating, not user slotted ones.

Oh, definitely, I understand that part, I just wonder if its worth it to create dynamic refs to internally-rendered elements based on slotted content.  Just not sure its worth creating all those refs in memory when a DOM query suffices.  Do you now which approach is likely to be more performant?

Reply:
It would be nice to refactor this to store a reference to these chip elements created by the component rather than setting a unique id on them. Then the internal shadow DOM doesn't need to be queried every time `refreshSelectionDisplay()`is called. (which happens on resize observer call and `calciteInternalComboboxItemChangeHandler`)

I'm pretty sure its more performant to just have a reference pointer to the element rather than having to do lookups through the dom every time these are called.

We can do this as a follow up issue since it is existing code. @Elijbet can you create a refactor issue?

Reply:
[Refactor dynamically created chips to use internal references instead of DOM queries](https://github.com/Esri/calcite-design-system/issues/12052#top)
#12052